cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactCodec.java

```diff
@@ -255,6 +255,24 @@ public long getTimestamp(byte[] rowKey, byte[] column) {
     return timebase + leftover;
   }
 
+  static byte[][] getSplits(int aggGroupsCount) {
+    // Row key format:
+    // <version><encoded agg group><time base>...
+    // Version is fixed. We assume agg group name is encoded with 1, ..., <aggGroupsCount>.
+    int encodedIdSize = EntityTable.computeSize();
+    int rowKeySize = VERSION.length + encodedIdSize;
+    // NOTE: we don't need to include first split, it will be added automatically (e.g. by HBase).
+    byte[][] splits = new byte[aggGroupsCount - 1][];
+    for (int i = 2; i <= aggGroupsCount; i++) {
```

Comment:
we start from 2, as the first split is added by hbase ? if so, would the first aggregation would get matched in the first split ?